Add gating tests to nightly

dhnunes opened the pull request:
The objective of this change is to address the problem mentioned in this
thread: https://lists.fedorahosted.org/archives/list/freeipa-devel@lists.fedorahosted.org/message/FIOWT53LJMAZQYHOTT4BEAJX5Q6422LB/

Since the concept of nightly is being a superset of gating, the gating
tests are incorporated in nightly in this commit.

Signed-off-by: Diogo Nunes <diogo.hap@gmail.com>

tiran commented:
* Travis CI will fail. You have to rebase this PR when https://github.com/freeipa/freeipa/pull/2602 has landed.
* We generally keep both nightly yaml files in sync. Please also update the rawhide yaml file.

flo-renaud commented on ipatests/prci_definitions/nightly_master.yaml:
```diff
@@ -38,6 +38,222 @@ jobs:
+      class: RunPytest
+      args:
+        build_url: '{fedora-28/build_url}'
+        test_suite: test_integration/test_caless.py::TestServerReplicaCALessToCAFull
```
This test is already present in gating as `fedora-28/test_caless_TestServerReplicaCALessToCAFull`.

flo-renaud commented on ipatests/prci_definitions/nightly_master.yaml:
```diff
@@ -38,6 +38,222 @@ jobs:
+      class: RunPytest
+      args:
+        build_url: '{fedora-28/build_url}'
+        test_suite: test_integration/test_replica_promotion.py::TestSubCAkeyReplication
```
This test is already present as `fedora-28/test_replica_promotion_TestSubCAkeyReplication`

flo-renaud commented on ipatests/prci_definitions/nightly_master.yaml:
```diff
@@ -38,6 +38,222 @@ jobs:
+      class: RunPytest
+      args:
+        build_url: '{fedora-28/build_url}'
+        test_suite: test_integration/test_dnssec.py::TestInstallDNSSECFirst
```
This test is already included in `fedora-28/test_dnssec`.

netoarmando commented:
Blocked by PR #2605.

netoarmando commented:
Related to this: https://pagure.io/freeipa/issue/7743

netoarmando commented on ipatests/prci_definitions/nightly_rawhide.yaml:
```diff
@@ -33,11 +33,191 @@ jobs:
-        template: &ci-master-frawhide
-          name: freeipa/ci-master-frawhide
+        template: &ci-master-rawhide
+          name: freeipa/ci-master-rawhide
```
Why change `frawhide` to `rawhide`? The only available template is `freeipa/ci-master-frawhide`: https://app.vagrantup.com/freeipa/boxes/ci-master-frawhide

dhnunes replied:
Was a typo, rebased thanks!

flo-renaud commented:
Hi @dhnunes 
thanks for the updated PR. With recent commits, there is a test `test_ntp_options` in nightly_master.yaml (with a wrong label  fedora-28/test_ntp_options that should be replaced with fedora-29/test_ntp_options) and in nightly_rawhide.yaml, but the same test is missing in nightly_f28.yaml.

dhnunes commented:
Hi @flo-renaud.

Just rebased to add new ```test_ntp_options``` in all three nightly definitions and fixed the wrong label fedora-28 to fedora-29 in nightly_master.

f-trivino commented on ipatests/prci_definitions/nightly_f28.yaml:
```diff
@@ -931,6 +1111,18 @@ jobs:
         timeout: 3600
         topology: *master_1repl
 
+    fedora-28/test_ntp_options:
```
indentation? :)

tiran commented:
re-run, COPR was down for maintenance.

flo-renaud commented:
@dhnunes 
thanks, LGTM.

dhnunes commented:
master:

* 03edd82d1ae8bb8c33eae3c18133e612432b9b67 PR-CI: Add gating tests to nightly_[master, f28, rawhide]
